packages/@aws-cdk/cx-protocol/lib/protocol.ts

```diff
@@ -0,0 +1,60 @@
+import * as fs from 'fs';
+import { ArtifactMetadataEntryType, ArtifactType, AssemblyManifest } from './assembly-manifest';
+
+/**
+ * Protocol utility class.
+ */
+export class Manifest {
+
+    /**
+     * Save manifest to to file.
+     *
+     * @param manifest - manifest.
+     */
+    public static save(manifest: AssemblyManifest, filePath: string) {
+        fs.writeFileSync(filePath, JSON.stringify(manifest, undefined, 2));
+    }
+
+    /**
+     * Load manifest from file.
+     */
+    public static load(filePath: string): AssemblyManifest {
+        const manifest: AssemblyManifest = JSON.parse(fs.readFileSync(filePath, 'UTF-8'));
+        Manifest.patchStackTags(manifest);
```

Comment:
Can this be done in the CLI instead of here?

Reply:
Yeah I mentioned in the PR description that I don't mind moving this away from here. Did you see my reasoning on why to put it here?

> This method loads a manifest from a given json file. Strictly speaking, it isn't required to enforce the compatibility we talked about. But I like the symmetry it adds. It also enforces AssemblyManifest compatibility as an output value, which seems reasonable as well because customers are theoretically exposed to it. Im also ok with moving it if there are strong objections.

Reply:
We decide to leave `load` here and provide both input and output compatibility postures.
As far as the `patchStackTags` goes, if we actually need to do this, i would rather it be here because otherwise the `load` function actually produces a corrupt interface. 
Im trying to get rid of this all together by just using `any` like you suggested. Resolving this discussion in favor of: https://github.com/aws/aws-cdk/pull/6724#discussion_r393932719